modules/core/src/bitgo.ts

```diff
@@ -427,848 +699,920 @@ const BitGo = function(params) {
-  const e = new Error();
-
-  // Kick off first load of constants
-  this.fetchConstants({}, function(err) {
-    if (err) {
-      // make sure an error does not terminate the entire script
-      console.error('failed to fetch initial client constants from BitGo');
-      debug(e.stack);
-    }
-  });
-};
-
-/**
- * Calculate the HMAC for the given key and message
- * @param key {String} - the key to use for the HMAC
- * @param message {String} - the actual message to HMAC
- * @returns {*} - the result of the HMAC operation
- */
-BitGo.prototype.calculateHMAC = function(key, message) {
-  return crypto.createHmac('sha256', key).update(message).digest('hex');
-};
-
-/**
- * Create a basecoin object
- * @param coinName
- */
-BitGo.prototype.coin = function(coinName): BaseCoin {
-  return GlobalCoinFactory.getInstance(this, coinName);
-};
+  /**
+   * Calculate the HMAC for the given key and message
+   * @param key {String} - the key to use for the HMAC
+   * @param message {String} - the actual message to HMAC
+   * @returns {*} - the result of the HMAC operation
+   */
+  calculateHMAC(key: string, message: string): string {
+    return crypto.createHmac('sha256', key).update(message).digest('hex');
+  }
 
-/**
- * Create a basecoin object for a virtual token
- * @param tokenName
- */
-BitGo.prototype.token = function(tokenName, callback) {
-  return co(function *() {
-    yield this.fetchConstants();
-    return this.coin(tokenName);
-  }).call(this).asCallback(callback);
-};
+  /**
+   * Create a basecoin object
+   * @param coinName
+   */
+  coin(coinName: string): BaseCoin {
+    return GlobalCoinFactory.getInstance(this, coinName);
+  }
 
-BitGo.prototype.getValidate = function() {
-  return this._validate;
-};
+  /**
+   * Create a basecoin object for a virtual token
+   * @param tokenName
+   * @param callback
+   */
+  token(tokenName: string, callback?: NodeCallback<BaseCoin>): Bluebird<BaseCoin> {
+    return co(function *() {
+      yield this.fetchConstants();
+      return this.coin(tokenName);
+    }).call(this).asCallback(callback);
+  }
 
-BitGo.prototype.setValidate = function(validate) {
-  if (!_.isBoolean(validate)) {
-    throw new Error('invalid argument');
+  /**
+   *
+   */
+  getValidate(): boolean {
+    return this._validate;
   }
-  this._validate = validate;
-};
 
-// Return the current BitGo environment
-BitGo.prototype.getEnv = function() {
-  return this.env;
-};
+  /**
+   *
+   */
+  setValidate(validate: boolean): void {
+    if (!_.isBoolean(validate)) {
+      throw new Error('invalid argument');
+    }
+    this._validate = validate;
+  }
 
-BitGo.prototype.clear = function() {
-  // TODO: are there any other fields which should be cleared?
-  this._user = undefined;
-  this._token = undefined;
-  this._refreshToken = undefined;
-  this._ecdhXprv = undefined;
-};
+  /**
+   * Return the current BitGo environment
+   */
+  getEnv(): EnvironmentName {
+    return this._env;
```

Comment:
Now that ```env``` has become ```_env```, wont this possibly mess up customers who might be using that?

Reply:
I thought of that, so `env` still exists, but it's deprecated.

See https://github.com/BitGo/BitGoJS/pull/488/files/8b35edfb3130a0b3df4db84ac359ea195d4abf3c#diff-b2b93516e1aec818bf25c684902edc17R480 and https://github.com/BitGo/BitGoJS/pull/488/files/8b35edfb3130a0b3df4db84ac359ea195d4abf3c#diff-b2b93516e1aec818bf25c684902edc17R383